### ISSUE TITLE
'reflect-metadata' requirement on nodejs

### ISSUE THREAD

**Issue by timfish** [aurelia/store#65]
The following code, crashes in nodejs with `TypeError: Reflect.getOwnMetadata is not a function`
```
import { Store } from 'aurelia-store';

const store = new Store({});
```
This is because `Store` has the `@autoinject` decorator [on it](https://github.com/aurelia/store/blob/705db3dac71d95b1fec526b14cff4c6964f93b05/src/store.ts#L38):
```
@autoinject()
export class Store<T> {
  public readonly state: Observable<T>;
```
Is this decorator required here?

**Comment by zewa666** [comment]
A way to fix this would be `npm install --save-dev reflect-metadata` and include `node_modules/reflect-metadata/Reflect.js`.

**Comment by zewa666** [comment]
Now that you're asking though, currently the autoinject serves no purpose as nothing gets constructor injected. Nevertheless, chances are there that at a later point a decorator with metadata dependency will be required, so checking whether above fix from my comment helps, would still be good to know.

**Comment by timfish** [comment]
Importing it at the top of main entry point does fix it:
```
import 'reflect-metadata';
...
```
I suppose the real issue is that `aurelia-metadata` requires `Reflect.*` to be polyfilled but doesn't list `reflect-metadata` as a dependency.

**Comment by zewa666** [comment]
that's true and its also not required for Babel, since it brings most polyfills by itself. So it was more the fact that TS used to be a step-child so far but will definitely improve with vNext.

Are there any action points you see that should be taken or can we close this issue?
